sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/DiagnosticExtensions.cs

```diff
@@ -113,7 +129,7 @@ private static void AddLinkedDiagnostics(this DiagnosticScope scope, IReadOnlyDi
 
         private static void AddLinkedDiagnostics(this DiagnosticScope scope, PropertiesMap properties)
```

Comment:
What's the scenario where we'd be operating on a `Microsoft.Azure.Amqp.AmqpMessage` instance?   I thought we did all the diagnostic manipulation at the model level.

Reply:
We do, but with the changes to cache the AmqpMessage when using a batch, we needed a way to extract the diagnostic Id from the message in order to add the link to the sender activity before sending.